examples/adjoints/adjoints_ex5/adjoints_ex5.C

```diff
@@ -498,12 +498,24 @@ int main (int argc, char ** argv)
           // Solve the forward problem at time t, to obtain the solution at time t + dt
           system.solve();
 
-          // Output the H1 norm of the computed solution
-          libMesh::out << "|U("
-                       << system.time + system.deltat
-                       << ")|= "
-                       << system.calculate_norm(*system.solution, 0, H1)
-                       << std::endl;
+          if(param.timesolver_tolerance)
+          {
+            // Output the H1 norm of the computed solution
+            libMesh::out << "|U("
+                         << system.time + dynamic_cast<AdaptiveTimeSolver &>(*(system.time_solver)).completedtimestep_deltat
```

Comment:
This is pretty hideous code.  Could we add a virtual accessor to TimeSolver, perhaps?  TimeSolver::last_deltat() would default to "return system.deltat", but AdaptiveTimeSolver would return completedtimestep_deltat?